### DISCUSSION TITLE
v1.7.4273 update made vscode slow in mac

### DISCUSSION THREAD

**Discussion by saravanapriyanm** [community/community#8861] [Copilot Conversations] [answered]
Last update made most things in VS code slow.
- Quick search
- Source control changes viewer
- Save file

Fix:
Disabling extension or installing previous version works.

**Comment by pbr1111** [comment]
It's also happening in VSCode for Windows.

**Comment by hsenag** [comment]
Hi @pbr1111 and @saravanapriyanm, I'm sorry you're hitting this. We've so far been unable to reproduce it, unfortunately.

We do have a more frequent series of releases of the [GitHub Copilot Nightly](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-nightly) extension.

I wonder if you'd mind trying to identify when it started happening in that series? It would help us narrow down which change in our code might have triggered it.

**Reply by luizeboli** [reply]
From my tests, nightly version 1.7.4188 is working well. As soon as I install 1.7.4230 (or newer ones) things starts to get really slow (saving a file, searching, github extension is also slow)

Using 1.7.4188 version I got this in system monitor:
![image](https://user-images.githubusercontent.com/13091635/146207111-d1d68970-3dc7-4fae-8089-c230d83244c9.png)

Using 1.7.4230 and trying to save a file this process jumps to 100%:
![image](https://user-images.githubusercontent.com/13091635/146207285-7f351f6e-84cd-45e0-a6e6-b9f94f61377b.png)

When I kill this process:
![image](https://user-images.githubusercontent.com/13091635/146207727-e31a835a-2095-43ae-a15e-49598acef789.png)

**Reply by hsenag** [reply]
@luizeboli thanks very much for doing that! Would you mind trying the latest "nightly" release, 1.7.4348 (or any later one)?

It's a bit of a shot in the dark as we can't reproduce the problem yet, but we spotted one place where it's possible our code could loop infinitely and have put in a guard to prevent that happening.

**Reply by beyerleinf** [reply]
Hi @hsenag, I'm still seeing the same behaviour, unfortunately. For me the issue comes up as soon as I install 1.7.4324. 1.7.4288 is still working fine.

**Reply by luizeboli** [reply]
Hi @hsenag, I couldn't reproduce the problem anymore. Seems solved for me. I'll test a little bit more and gonna update this reply probably by tomorrow

---------

edit:

Everything is running smoothly using 1.7.4348

**Reply by pbr1111** [reply]
In my case, it has not been fixed in 1.7.4348 nor in 1.7.4350.

**Comment by dBiazioli** [comment]
The same thing is happening on VSCode for Linux.

Looks like some process never finishes, which start making things slow... I checked with the System Monitor, and even if no code is written, one core of my cpu stays at 100% when Copilot is enables

**Comment by Krzysztof-Cieslak** [comment]
@saravanapriyanm @dBiazioli @pbr1111, 

Thanks for the report, we're working on reproducing this.

A couple of questions:
* Does Copilot work "normally" for you when you experience this problem? I.e. do you get suggestions when you type some code?
* Is it happening on any specific files? Any particular programming language? 
* Could you maybe try to provide us with some reproduction that triggers this issue? Is there any set of actions that triggers it, or is it just happening while you code?

**Reply by dBiazioli** [reply]
- No, on my case Copilot only shows the 'experiencing temporary issues' message
- I got the issue on different files (Gherkin, JS, TS)
- On my case, multiple actions triggered the issue, mostly background processes...

**Reply by pbr1111** [reply]
- No, it simply does not initialize and delays the initialization of the other extensions. 
- It's happening on all files. It is a repository that contains different project types (.NET and React). I have tried deleting the .git folder and do a git init, it seems to work fine. The 2 projects where it happens to me have precommit hooks with husky, but I have tried with github projects of examples with husky and I can't reproduce it. It probably has nothing to do with git, but it does seem to. 
- It happens when the vscode starts and when I save a file.

**Comment by Artemixed** [comment]
Heyo I'm having this exact same issue! VSC got super slow and github copilot tells me that it's experiecing temporary issues

Windows 10 Pro x64 version 10.0.22000 build 22000
VS Code 1.63
Copilot 1.7.4273

**Comment by August-Rushme** [comment]
Hi ,I also had this problem, resulting in a very high CPU utilization. At first, I thought it was the problem of vscode, but after the Extension Bisect investigation, it was found that it was the problem of GitHub copilot (v1.7.4273). Go back to the previous version to solve the problem.

**Comment by MassivDash** [comment]
The same issue: 100% CPU utilisation breaking everything in VSCode on MacOs system.

**Comment by vateseeb** [comment]
I experienced the same issue. Git changes, file saving and typescript features where not working. Couldn't reproduce it with a new workspace only containing a few files but opening an existing one almost always caused this issues. Switched to version v1.7.4016 now which fixed it for me.

Windows 10 Enterprise x64 10.0.19042 Build 19042
VSCode 1.63.1
Copilot 1.7.4273

**Comment by hsenag** [comment]
Hi,

I'm sorry that this is still causing trouble for some people. Unfortunately we've still been unable to reproduce it :-(

Please could you try 1.7.4386 (nightly) or 1.7.4395 (stable) and confirm if you still have the problem?

These versions contain the same fix as in 1.7.4348, which one user reported fixed the problem and other users reported didn't.

If you do still have the problem, if you are able to help with any of the following debugging steps, we'd be very grateful:

- If you are able to share the project you are working on (e.g. if it's a public GitHub repo), please let us know, either here or by email to the addresses listed below. It might help us reproduce it.
- We have added two debug flags to disable possibly problematic recent changes:
  - "debug.limitPrompts"
  - "debug.ignoreRepo"

  For each of these flags _separately_, set them to true in the extension settings:
    - Go to the [extension settings](https://github.com/github/copilot-docs/blob/main/docs/visualstudiocode/configuring.md#settings-for-the-extension).
    - Under "GitHub > Copilot: Advanced", select ["Edit in settings.json"](https://user-images.githubusercontent.com/451141/146527031-4cda24fd-af0f-4101-b594-0c7d767ee5a0.png).
    - Add a "github.copilot.advanced" section and [set one of the flags to `true`](https://user-images.githubusercontent.com/451141/146527606-cc40a7a6-bb92-4385-b447-99199bcb7ad9.png).
    - Try to reproduce the problem. Let us know if either of the flags does fix the problem, or if neither of them does.
- [Collect a CPU profile from VSCode](https://github.com/microsoft/vscode/wiki/Performance-Issues#profile-the-running-extensions) while the problem is happening and email it to us.
  
You can contact us directly via email: hsenag@github.com and krzysztof-cieslak@github.com

Regards,

Ganesh

**Reply by vateseeb** [reply]
It seems like the upgrade to 1.7.4395 fixed it for me as well. Even without setting the debug flags I couln't reproduce the issue so far. 
Thanks for the quick solution Ganesh!

**Reply by pbr1111** [reply]
In my case, the only way to solve the problem is by enabling __debug.ignoreRepo__

**Reply by frittsy** [reply]
Fixed me for me now in `1.7.4395` without using the debug flags. Thank you!

**Comment by dBiazioli** [comment]
Upgraded to `v1.7.4395` and seems to be solved for me too!

**Comment by pbr1111** [comment]
I finally found a way to reproduce my bug on Windows. I use Node.js 16.13.1 (I have tried also with Node.js 17.2.0) and the latest version of Git for Windows (2.34.1).

To reproduce the problem, you need to create a repo with a js file. You can use these commands in a bash console: 
```sh
mkdir example-repo && cd example-repo && git init && touch test.js
```

Overwrite the `.git/config` file with the following content:
```
[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[remote "origin"]
	url = https://largelargeorganizationname@dev.azure.com/largelargeorganizationname/largelargelargeprojectname/_git/largelargelargerepo
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
```
Now open vscode, add some content to the test.js file and save. 

I have been debugging the extension.js file of the extension. It gets stuck in the `fs.promises.readFile` of the `.git/config` file. I think you should consider using the API that exposes the [official VSCode Git extension](https://github.com/microsoft/vscode/tree/main/extensions/git), since it's probably entering into a reading conflict and it will help you to remove some hacks that you use in this part of the code.

**Reply by hsenag** [reply]
Hi @pbr1111,

We really appreciate the detailed debugging, thank you!

We believe we've identified a regex that was "suboptimal" 😉  from your report and we've put out a quick fix that we hope addresses that (1.7.4423 - nightly/1.7.4421 - stable). We'll take a more careful look after the weekend to double-check that it really fixes your repro case. 

We also appreciate your point about the `readFile` and will take a close look at improving that.

Regards,

Ganesh

**Reply by pbr1111** [reply]
Hi @hsenag, I just tried the new version and it solved the issue 😄 Thanks for your efforts!

**Comment by hsenag** [comment, marked as answer]
We believe that the problem is now fixed in the most recent releases (1.7.4423 - nightly/1.7.4421 - stable). Many thanks to all the people who reported the symptoms, tracked down which versions introduced it and tried out new versions. Special thanks to @pbr1111 for taking us right to the precise cause.

**Reply by sag1v** [reply]
Is this only a windows fix? I'm on mac and having similar issues.
Seems like i'm on the latest version.

Version: 1.63.2
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:37:28.172Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0

**Reply by hsenag** [reply]
It should have applied to all platforms. What version of the Copilot addin do you have?

**Reply by sag1v** [reply]
Aw is this only a copilot related issue? I don't use it (AFAIK)

**EDIT** 
I still need to get used to these threads in github, sorry :)

**Reply by hsenag** [reply]
Yes, it's copilot-specific :-) VSCode is owned by a team in Microsoft, their guide on raising issues is here: https://github.com/Microsoft/vscode/blob/main/CONTRIBUTING.md#reporting-issues

**Comment by airtonix** [comment]
still using a billion percent of my cpu and my GPU for some reason???

**Comment by AleksMeshkov** [comment]
Same thing, I was working in a caffe with no charger on. It drained the battery

**Comment by paulusm** [comment]
Getting this periodically. Manjaro Linux VS Code 1.18.1 Copilot 1.104.341 Runaway process is Zygote > gpu-process. At the same time the status bar Copilot icon turns into a spinner.

**Reply by grzegorz-herman** [reply]
I have the same issue. VS Code 1.83.0-insider (commit: f27dd75c3870a63c84a7976a6568655eca59d612), Copilot v1.114.428, also runaway gpu-process and spinner. Triggering Copilot suggestions by typing some code usually works and sometimes makes the problem go away, until it starts again, seemingly randomly.
